src/node/node_state.h

```diff
@@ -477,8 +467,6 @@ namespace ccf
               sm.advance(State::partOfNetwork);
             }
 
-            join_timer.reset();
```

Comment:
How does the new timer implementation know when to stop sending join requests?

Reply:
In the callback code we re-schedule the timer if required. Unless i missed something?